### ISSUE TITLE
help request: Route cannot support both http and h2c

### ISSUE THREAD

**Issue by wangfeng22** [apache/apisix#8241]
### Description

when I set node_listen as follows, I can't access other upstreams by http.
```
apisix:
  node_listen:                      # This style support multiple ports
    - 80
       enable_http2: true
```
`curl http://domain` return `???`

### Environment

- APISIX version (run `apisix version`): 2.13.2
- Operating system (run `uname -a`):
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`):
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

**Comment by Hazel6869** [comment]
@wangfeng22  Hi, may be you should do like this. https://github.com/apache/apisix/blob/e523132f35fe2eb7e1d06990d037f18f89c9dcba/conf/config-default.yaml#L103

**Comment by wangfeng22** [comment]
> @wangfeng22 Hi, may be you should do like this.
> 
> https://github.com/apache/apisix/blob/e523132f35fe2eb7e1d06990d037f18f89c9dcba/conf/config-default.yaml#L103

what's the difference?

**Comment by tokers** [comment]
@wangfeng22 Yes, the H2C mechanism is not supported by Apache APISIX. For using HTTP/2, either through TLS ALPN or NPN (which means you have to use TLS), or use the [HTTP/2 Connection Preface](https://httpwg.org/specs/rfc7540.html#ConnectionHeader).

**Comment by wangfeng22** [comment]
> @wangfeng22 Yes, the H2C mechanism is not supported by Apache APISIX. For using HTTP/2, either through TLS ALPN or NPN (which means you have to use TLS), or use the [HTTP/2 Connection Preface](https://httpwg.org/specs/rfc7540.html#ConnectionHeader).

Thank you very much. @tokers
